pkg/operator/eventwatch/glob.go

```diff
@@ -0,0 +1,61 @@
+package eventwatch
+
+import (
+	"strings"
+)
+
+// reasonMatch will test a string pattern, potentially containing globs, against a
+// subject string. The result is a simple true/false, determining whether or
+// not the glob pattern matched the subject text.
+//
+// Credit: https://github.com/ryanuber/go-glob
+func reasonMatch(pattern, reason string) bool {
+	if pattern == "" {
+		return reason == pattern
+	}
+
+	// If the pattern _is_ a glob, it matches everything
+	if pattern == "*" {
+		return true
+	}
+
+	// If the pattern does not contain a glob, require strict match by default
+	if !strings.Contains(pattern, "*") {
+		return pattern == reason
+	}
+
+	parts := strings.Split(pattern, "*")
+
+	if len(parts) == 1 {
+		// No globs in pattern, so test for equality
+		return reason == pattern
+	}
+
+	leadingGlob := strings.HasPrefix(pattern, "*")
+	trailingGlob := strings.HasSuffix(pattern, "*")
+	end := len(parts) - 1
+
+	// Go over the leading parts and ensure they match.
+	for i := 0; i < end; i++ {
+		idx := strings.Index(reason, parts[i])
```

Comment:
the case of `*foo` seems to go through fine, but is full of edge cases. Maybe make that explicit. Easier to read.

Reply:
this is copied from the original repo and it is tested with unit test... i don't want to hack this up too much

Reply:
I was impressed by the clever algorithm. Now you steal me my amazement :D